### PR TITLE
Warnint 64to32 6186 v19.5

### PR DESCRIPTION
**Pull request by catenacyber**
[Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6186

Describe changes:
- fix `-Wshorten-64-to-32` warnings for some files (a*)

First commit of #9840
#10899 with needed rebase

There must be some stuff to review more carefully

### CODE REVIEW

File: src/app-layer-frames.c [hunk 1]
```diff
@@ -368,7 +368,9 @@ static int FrameSlide(const char *ds, Frames *frames, const TcpStream *stream, c
     }
     frames->cnt = x;
     uint64_t o = STREAM_BASE_OFFSET(stream) + slide;
-    frames->left_edge_rel = le - (STREAM_BASE_OFFSET(stream) + slide);
+    BUG_ON(o > le);
```
Comment (victorjulien) [inline src/app-layer-frames.c:371]: should these be DEBUG_VALIDATE_BUG_ON?

File: src/app-layer-htp-range.h [hunk 2]
```diff
@@ -40,7 +40,7 @@ typedef struct HttpRangeContainerBuffer {
     /** offset of bytes written in buffer (relative to the start of the range) */
     uint64_t offset;
     /** number of gaped bytes */
-    uint64_t gap;
+    uint32_t gap;
```
Comment (victorjulien) [inline src/app-layer-htp-range.h:43]: a file can be larger than 4GiB, so a gap in it's data could also exceed 4GiB

File: src/app-layer-htp-range.h [hunk 3]
```diff
@@ -63,7 +63,7 @@ typedef struct HttpRangeContainerFile {
     /** key length */
     uint32_t len;
     /** expire time in epoch */
-    uint32_t expire;
+    uint64_t expire;
```
Comment (victorjulien) [inline src/app-layer-htp-range.h:66]: `SCTime`?

**Review by victorjulien — CHANGES_REQUESTED** [review]
Some feedback and suggestions inline.

File: src/app-layer-frames.c [hunk 4]
```diff
@@ -790,15 +791,16 @@ static void FramePrune(Frames *frames, const TcpStream *stream, const bool eof)
         }
     }
     frames->cnt = x;
-    frames->left_edge_rel = le - STREAM_BASE_OFFSET(stream);
+    BUG_ON(le < STREAM_BASE_OFFSET(stream));
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/app-layer-frames.c:794]: This one was your code :
> git blame -v src/app-layer-frames.c | grep BUG_ON

`1556e86c7db (Victor Julien    2021-12-03 07:40:56 +0100 801)     BUG_ON(le < STREAM_BASE_OFFSET(stream));`

File: src/app-layer-frames.c (hunk 1 above)
Comment (catenacyber) [reply src/app-layer-frames.c:371]: ok

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20970

File: src/app-layer-htp-range.h (hunk 2 above)
Comment (catenacyber) [reply src/app-layer-htp-range.h:43]: ok, will saturate it...

File: src/app-layer-htp-range.h (hunk 3 above)
Comment (catenacyber) [reply src/app-layer-htp-range.h:66]: nice

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/11257

**Pull request closed without merging**